### PR TITLE
SDK X Unity 10.3.0 Changes

### PR DESCRIPTION
**Pull request by jaydeep-helpshift**

### CODE REVIEW

**Review by hs-chetan-urkude — APPROVED** [review]

File: docs/sdkx-unity/getting-started-android.mdx [hunk 1]
```diff
@@ -41,17 +41,16 @@ Guide to integrating the Unity plugin for the Helpshift SDK X which you can call
...
 
 <div className="text-sm compact ">
 
-|                                                                      |                                                              |
-| -------------------------------------------------------------------- | ------------------------------------------------------------ |
-| helpshiftX-plugin-unity-[% sdkxunity.package_version %].unitypackage | Unity package of Helpshift SDK                               |
-| unity-jar-resolver (v1.2.170.0)                                      | Resolves Android Helpshift package support lib dependencies. |
+|                                               |             |
+| ----------------------------------------------|-------------|
+| helpshiftX-plugin-unity-v10.3.0.unitypackage | Unity package of HelpshiftX SDK         |
```
Comment (hs-chetan-urkude) [inline docs/sdkx-unity/getting-started-android.mdx:65]: `HelpshiftX` This is type'o or intentional ?

Comment (sumeet-helpshift) [reply docs/sdkx-unity/getting-started-android.mdx:65]: this should be Helpshift SDK X

File: docs/sdkx-unity/delegates-android.mdx [hunk 2]
```diff
@@ -275,6 +328,86 @@ public class HSEventsListener : IHelpshiftEventsListener
...
+
+```java
+public class HSEventsListener : IHelpshiftEventsListener
+{
+  public void HandleHelpshiftEvent(string eventName, Dictionary<string, object> eventData)
+    if (eventName.Equals(HelpshiftEvent.AGENT_MESSAGE_RECEIVED)) {
+            String publishId = (String) eventData[HelpshiftEvent.DATA_PUBLISH_ID];
+            String type = (String) eventData[HelpshiftEvent.DATA_MESSAGE_TYPE];
+            String body = (String) eventData[HelpshiftEvent.DATA_MESSAGE_BODY];
+            Long createdTs = (Long) eventData[HelpshiftEvent.DATA_CREATED_TIME];
```
Comment (sumeet-helpshift) [inline docs/sdkx-unity/delegates-android.mdx:371]:
```suggestion
            long createdTs = (long) eventData[HelpshiftEvent.DATA_CREATED_TIME];
```

File: docs/sdkx-unity/delegates-android.mdx [hunk 3]
```diff
@@ -275,6 +328,86 @@ public class HSEventsListener : IHelpshiftEventsListener
...
+
+            // Utils.isEmpty() is null and empty Check
+            if (Utils.isEmpty(publishId) && Utils.isEmpty(type) && Utils.isEmpty(body) && createdTs == null) {
+              Debug.Log(TAG, "Received no data");
+              return;
+            }
+
+            Debug.Log("publishId " + publishId + " type " + type + " body " + body + " createdTs " + createdTs);
+            List<Object> attachments = (List<Object>) eventData.get(HelpshiftEvent.DATA_ATTACHMENTS);
```
Comment (sumeet-helpshift) [inline docs/sdkx-unity/delegates-android.mdx:380]:
```suggestion
            List<object> attachments = (List<object>)eventData[HelpshiftEvent.DATA_ATTACHMENTS];
```

File: docs/sdkx-unity/delegates-android.mdx [hunk 4]
```diff
@@ -275,6 +328,86 @@ public class HSEventsListener : IHelpshiftEventsListener
...
+
+            if (Utils.isEmpty(attachments)) {
+              Debug.Log("No attachments received in message");
+            } else {
+
+              for (int i = 0; i < attachments.size(); i++) {
+                Dictionary<String, Object> attachment = (Dictionary<String, Object>) attachments.get(i);
```
Comment (sumeet-helpshift) [inline docs/sdkx-unity/delegates-android.mdx:387]:
```suggestion
                Dictionary<String, object> attachment = (Dictionary<String, object>) attachments.get(i);
```

File: docs/sdkx-unity/delegates-android.mdx [hunk 5]
```diff
@@ -275,6 +328,86 @@ public class HSEventsListener : IHelpshiftEventsListener
...
+{
+  public void HandleHelpshiftEvent(string eventName, Dictionary<string, object> eventData)
+    if (eventName.Equals(HelpshiftEvent.AGENT_MESSAGE_RECEIVED)) {
+            String publishId = (String) eventData[HelpshiftEvent.DATA_PUBLISH_ID];
+            String type = (String) eventData[HelpshiftEvent.DATA_MESSAGE_TYPE];
+            String body = (String) eventData[HelpshiftEvent.DATA_MESSAGE_BODY];
+            Long createdTs = (Long) eventData[HelpshiftEvent.DATA_CREATED_TIME];
+
+            // Utils.isEmpty() is null and empty Check
+            if (Utils.isEmpty(publishId) && Utils.isEmpty(type) && Utils.isEmpty(body) && createdTs == null) {
```
Comment (sumeet-helpshift) [inline docs/sdkx-unity/delegates-android.mdx:374]:
```suggestion
            if (Utils.isEmpty(publishId) && Utils.isEmpty(type) && Utils.isEmpty(body) && createdTs == 0) {
```

File: docs/sdkx-unity/delegates-android.mdx [hunk 6]
```diff
@@ -275,6 +328,86 @@ public class HSEventsListener : IHelpshiftEventsListener
...
+                  Debug.Log("Received no data for attachment " + (i + 1));
+                  continue;
+                }
+
+                if (Utils.isEmpty(url) || Utils.isEmpty(fileName) || size == null) {
+                  Debug.Log("Received incomplete data for attachment " + (i + 1));
+                  continue;
+                }
+
+                Debug.Log(TAG, "Attachment No. : " + (i + 1) + ", url: " + url + ", contentType: " + contentType + ", fileName: " + fileName + ", size: " + size);
```
Comment (sumeet-helpshift) [inline docs/sdkx-unity/delegates-android.mdx:404]:
```suggestion
                Debug.Log("Attachment No. : " + (i + 1) + ", url: " + url + ", contentType: " + contentType + ", fileName: " + fileName + ", size: " + size);
```

File: docs/sdkx-unity/delegates-android.mdx [hunk 7]
```diff
@@ -275,6 +328,86 @@ public class HSEventsListener : IHelpshiftEventsListener
...
+              Debug.Log("No attachments received in message");
+            } else {
+
+              for (int i = 0; i < attachments.size(); i++) {
+                Dictionary<String, Object> attachment = (Dictionary<String, Object>) attachments.get(i);
+
+                String url = (String) attachment[HelpshiftEvent.DATA_URL];
+                String contentType = (String) attachment[HelpshiftEvent.DATA_CONTENT_TYPE];
+                String fileName = (String) attachment[HelpshiftEvent.DATA_FILE_NAME];
+                Integer size = (Integer) attachment[HelpshiftEvent.DATA_SIZE];
```
Comment (sumeet-helpshift) [inline docs/sdkx-unity/delegates-android.mdx:392]:
```suggestion
                int size = (int) attachment[HelpshiftEvent.DATA_SIZE];
```

File: docs/sdkx-unity/delegates-android.mdx [hunk 8]
```diff
@@ -275,6 +328,86 @@ public class HSEventsListener : IHelpshiftEventsListener
...
+
+                if (Utils.isEmpty(url) && Utils.isEmpty(fileName) && size == null) {
```
Comment (sumeet-helpshift) [inline docs/sdkx-unity/delegates-android.mdx:394]:
```suggestion
                if (Utils.isEmpty(url) && Utils.isEmpty(fileName) && size == 0) {
```

**Review by sumeet-helpshift — APPROVED** [review]

File: docs/sdkx-unity/release-notes-unity.mdx [hunk 9]
```diff
@@ -23,16 +23,48 @@ import {
...
+- Fixed a bug where `presentFullScreenOniPad` config was not being honored on opening SDK screen via notification click.
+- Removed bitcode support from SDK X framework. SDK X now ships in non-bitcode variant only.
+
+**Android**
+
+- Added support for Android 14.
+- SDK X now supports only Android 7, i.e API Level 24, and above. SDK will be non-operational when your app is installed on a device below Android 7 Nougat.
+- Moved to AndroidX support libraries. SDK X will not compile with legacy Android Support libraries anymore.
+- Added support for multiple attachments in conversations (Support for multiple attachment in bot upload step will be added in Q1`24).
+- Added a new API to fetch SDK X version. For more details, [refer here](/sdkx-unity/support-tools-android/#get-sdk-version).
```
Comment (sumeet-helpshift) [inline docs/sdkx-unity/release-notes-unity.mdx:54]:
```suggestion
- Added a new API to fetch SDK X version. For more details, [refer here](/sdkx-unity/support-tools-android/#getting-the-sdk-version).
```

File: docs/sdkx-unity/release-notes-unity.mdx [hunk 10]
```diff
@@ -23,16 +23,48 @@ import {
...
+- Added a new API to close active SDK session. For more details, [refer here](/sdkx-unity/support-tools-android/#close-session).
```
Comment (sumeet-helpshift) [inline docs/sdkx-unity/release-notes-unity.mdx:55]:
```suggestion
- Added a new API to close active SDK session. For more details, [refer here](/sdkx-unity/support-tools-android/#closing-an-open-sdk-session).
```

**Review by sumeet-helpshift — APPROVED** [review]

**Review by IamJoyson — APPROVED** [review]

File: docs/sdkx-unity/getting-started-android.mdx [hunk 11]
```diff
@@ -31,8 +31,22 @@ Guide to integrating the Unity plugin for the Helpshift SDK X which you can call
...
+* SDK X is functional only above API Level 24 i.e Android OS 7 Nougat and above.
+
+* The `minSDKVersion` in code is kept as API Level 16 to allow importing the SDK and compiling in apps that allow installation till API Level 16 i.e Android 4.4 Jelly Bean.
+
+* If your app is installed on a device below API level 24 (i.e below Android OS 7 Nougat) then you can choose to have a different medium for providing support. For example, Email, Web etc.
+
+<Admonition type="info" title="Note">
+
+- Update (August 2023): SDK X 10.2.x is functional only above API Level 24 i.e Android OS 7 Nougat and above.
+- SDK X 10.3.0 functions and is supported on Android 7 and above (API level 24 and above). You can still compile your app using a minimum API level of 16, but the support starts for API level 24 and above.
```
Comment (jaydeep-helpshift) [inline docs/sdkx-unity/getting-started-android.mdx:45]:
```suggestion
- SDK X 10.3.0 is supported on Android 7 and above (API level 24 and above). You can still compile your app using a minimum API level of 16, but the support starts for API level 24 and above.
```

File: docs/sdkx-unity/getting-started-android.mdx [hunk 12]
```diff
@@ -31,8 +31,22 @@ Guide to integrating the Unity plugin for the Helpshift SDK X which you can call
...
+  For details, please refer to [this article](https://support.helpshift.com/hc/en/13-helpshift-technical-support/faq/1217-sdk-x-os-technical-support-policy/).
+- Helpshift SDK X supports Android 14, but Unity currently lacks official support for this version. Developers using Unity may face compatibility issues when integrating SDKX with Android 14.
```
Comment (jaydeep-helpshift) [inline docs/sdkx-unity/getting-started-android.mdx:47]:
```suggestion
- Helpshift SDK X supports Android 14, but Unity currently lacks official support for this version. Developers using Unity may face compatibility issues when integrating SDK X with Android 14.
```

**Pull request merged**